### DISCUSSION TITLE
How do I verify if the VAD model is actually using my GPU power not the CPU power?

### DISCUSSION THREAD

**Discussion by JunYeNine** [snakers4/silero-vad#388] [Q&A] [answered]
Dear Alexander,

With the sample code in silero-vad.ipynb, how do I verify if the VAD model is actually using my GPU power not the CPU power?

#@title Install and Import Dependencies

# this assumes that you have a relevant version of PyTorch installed
!pip install -q torchaudio

SAMPLING_RATE = 16000

import torch
torch.set_num_threads(1)

from IPython.display import Audio
from pprint import pprint
# download example
torch.hub.download_url_to_file('https://models.silero.ai/vad_models/en.wav', 'en_example.wav')
USE_ONNX = False # change this to True if you want to test onnx model
if USE_ONNX:
    !pip install -q onnxruntime
  
model, utils = torch.hub.load(repo_or_dir='snakers4/silero-vad',
                              model='silero_vad',
                              force_reload=True,
                              onnx=USE_ONNX)

(get_speech_timestamps,
 save_audio,
 read_audio,
 VADIterator,
 collect_chunks) = utils

Kind regards,
Jun

**Comment by snakers4** [comment, marked as answer]
Hi,

The VAD is not designed to run on GPU.

**Reply by JunYeNine** [reply]
Thanks, that's good to know.

**Reply by deep410** [reply]
Can u provide more details around it @snakers4
